app/controllers/shelters_controller.rb

```diff
@@ -1,19 +1,36 @@
 class SheltersController < ApplicationController
+  include ActionView::Helpers::TextHelper
+
   def index
     @shelters = Shelter.all
   end
 
   def show
     @shelter = Shelter.find(params[:id])
     @reviews = @shelter.reviews
+    @pet_counter = @shelter.pets.count
+    @average_rating = @reviews.average(:rating)
+    @shelter_applications = Array.new
+    @shelter.pets.each do |pet|
+      @shelter_applications << PetApply.where(pet_id: pet.id).pluck(:apply_id)
+    end
+    @shelter_applications.flatten!
+    @numb_of_applications = @shelter_applications.count
   end
 
+
   def new
   end
 
   def create
-    Shelter.create(shelter_params)
-    redirect_to '/shelters'
+    shelter = Shelter.new(shelter_params)
+    if shelter.save
+      redirect_to '/shelters'
+    else
+      quantity = empty_fields(params).count
+      flash[:shelter_fields_notice] = "Shelter Creation Warning: You are missing #{pluralize(quantity, "field")}: #{empty_fields_convert}"
```

Comment:
Here is an opportunity to use that empty_fields on ANY page that has warning about required fields - We should consider moving this to application controller and updating